Specs-state (#290)
PrestaShop/prestashop-specs: content/1.7/back-office/international/locations/states-listing.md

---
title: States listing
weight: 1
---

> ## Bulk actions

Reviewer: Could you move this whole part before the Columns one ?

1. Refresh list

After clicking on it, the page is reloaded. Sorts and searches are kept.

2. Show SQL query

After clicking on it, the SQL query is displayed in a modal. The SQL query takes into account filters and sorts that have been done.

There are two buttons:

Close: After clicking on it, the modal is closed.
Export to SQL Manager: After clicking on it, the SQL query is opened in Advanced parameters > Database > SQL Manager.

3. Export to SQL Manager

After clicking on it, the SQL query is opened in Advanced parameters > Database > SQL Manager.

## Columns

1. ID

It's possible to sort by id asc or desc and search by id.

2. Name

It's possible to sort by name asc or desc and search by name.

3. ISO code

It's possible to sort by ISO code asc or desc and search by ISO code.

4. Zone

It's possible to sort by zone asc or desc and to search for a zone by selecting one in the drop-down.

5. Country

It's possible to sort by country asc or desc and search for a country by selecting one in the drop-down.

6. Enabled

It's possible to search for enabled or disabled countries by selecting either "yes" or "no" in the drop-down.

7. Actions

>  - **Edit** - The user can edit a feature as often as necessary, when pressed redirects to the edit page of the selected state.

Reviewer: ```suggestion
 - **Edit** - After clicking on it, the edit state page is displayed.
```

 
   👉 See the [specs of the add edit state]({{<ref "add-edit-state.md">}} "add edit state") 
   

>  - **Delete** - Deletes the selected state, after deletion shows the message: "Successful deletion.".

Reviewer: ```suggestion
 - **Delete** - After clicking on it, the selected state is deleted and the following success message is displayed: "Successful deletion.".
```

> ---

Reviewer: Same here, this file should be in the "states" folder: content/1.7/back-office/international/locations/states/states-listing.md

Reviewer: @Saimis777 I reopen this conversation because it's not fixed :)

title: States listing
weight: 1
---

# States

## Call to action

> A call to action buttont at the top right is displayed:

Reviewer: ```suggestion
A call to action button is displayed:
```

>  - **Add new state** - See the [specs of the add new state]({{<ref "add-new-state.md">}} "Add new state") 

Reviewer: I let you change the link after having renamed the file

> ### Settings wheel

Reviewer: ```suggestion
## Settings wheel
```

When clicking on the settings wheel, the following options are displayed:

> 1. Add new

Reviewer: ```suggestion
```

Reviewer: As for others migrated pages, this will be removed during the migration

> When pressed redirects to the [specs of the add new state]({{<ref "add-new-state.md">}} "Add new state") 

Reviewer: I let you change the link after having renamed the file

> When pressed redirects to the [specs of the add new state]({{<ref "add-new-state.md">}} "Add new state") 

Reviewer: ```suggestion
```

> 2. Refresh list

Reviewer: ```suggestion
1. Refresh list
```

After clicking on it, the page is reloaded. Sorts and searches are kept.

> 3. Show SQL query

Reviewer: ```suggestion
2. Show SQL query
```

After clicking on it, the SQL query is displayed in a modal. The SQL query takes into account filters and sorts that have been done.

There are two buttons:

Close: After clicking on it, the modal is closed.
Export to SQL Manager: After clicking on it, the SQL query is opened in Advanced parameters > Database > SQL Manager.

> 4. Export to SQL Manager

Reviewer: ```suggestion
3. Export to SQL Manager
```

After clicking on it, the SQL query is opened in Advanced parameters > Database > SQL Manager.

> 

Reviewer: ```suggestion
```

> The count of the states currently added is displayed on the left - "States - (current number of states added in the shop)".

Reviewer: ```suggestion
```

Reviewer: It's the same on all pages, no need to write it IMO

> ### Columns

Reviewer: ```suggestion
## Columns
```

1. ID

It's possible to sort by id asc or desc and search by id.

2. Name

It's possible to sort by name asc or desc and search by name.

3. ISO code

It's possible to sort by ISO code asc or desc and search by ISO code.

4. Zone

> This is a drop-down list of the continents that can be selected, and searched in the list.

Reviewer: ```suggestion
It's possible to sort by zone asc or desc and search zone by selecting one in the drop-down.
```

5. Country

> This is a drop-down list of the countries that can be searched.

Reviewer: ```suggestion
It's possible to sort by country asc or desc and search for a country by selecting one in the drop-down.
```

6. Enabled

> This is a drop-down list for selecting either "yes" or "no" can be searched.

Reviewer: ```suggestion
It's possible to search for enabled or disabled countries by selecting electing either "yes" or "no" in the drop-down.
```

7. Actions

>  - **Search** - Button for searching in the list.

Reviewer: ```suggestion
```

>  - **Edit** - When pressed redirects to the edit page of the selected state.

Reviewer: This could be improved, please look at what has been done in the previous spec: features-listing for example.

>  - **Delete** - Deletes the selected state, after deletion shows the message: "Successful deletion.".

Reviewer: This could be improved, please look at what has been done in the previous spec: features-listing for example.

> ### Bulk actions

Reviewer: Could you move this whole part bedore the Columns one ?

> ### Bulk actions

Reviewer: ```suggestion
## Bulk actions
```

Clicking on the _Bulk actions_ button allows performing bulk actions for the selected states (using the checkboxes). A user can:

- Select all/Unselect all: this button will select/unselect all the checkboxes at the beginning of each row of the listing.
- Enable selection - enables the selected state.
- Disable selection - disables the selected state.
- Delete selected: This button will delete the selected state. 
After clicking on "Delete selected", a modal with the title "Delete selection" is opened asking to confirm or cancel the action: "Are you sure you want to delete the selected item(s)?" (See issue [#14462](https://github.com/PrestaShop/PrestaShop/issues/14462)). The action can be canceled by clicking on the cross or on the cancel button.
After clicking on "Delete" button, the selected state gets deleted, a successful message is displayed "The selection has been successfully deleted."

> - Assign to a new zone - Selected states are assigned to a new zone.

Reviewer: ```suggestion
- Assign to a new zone - This buttons is currently not working, see https://github.com/PrestaShop/PrestaShop/issues/9971
```
